src/Mod/BIM/bimcommands/BimLayers.py

```diff
@@ -224,48 +220,45 @@ def accept(self):
 
             # label
             label = self.model.item(row, 1).text()
-            if label:
-                if obj.Label != label:
-                    if not changed:
-                        FreeCAD.ActiveDocument.openTransaction("Layers change")
-                        changed = True
-                    obj.Label = label
+            # Setting Label="" is possible in the Property editor but we avoid it here:
+            if label and obj.Label != label:
+                if not changed:
+                    FreeCAD.ActiveDocument.openTransaction("Layers change")
+                    changed = True
+                obj.Label = label
 
             # line width
             width = self.model.item(row, 2).data(QtCore.Qt.DisplayRole)
-            if width:
-                if obj.ViewObject.LineWidth != width:
-                    if not changed:
-                        FreeCAD.ActiveDocument.openTransaction("Layers change")
-                        changed = True
-                    obj.ViewObject.LineWidth = width
+            # Setting LineWidth=0 is possible in the Property editor but we avoid it here:
+            if width and obj.ViewObject.LineWidth != width:
+                if not changed:
+                    FreeCAD.ActiveDocument.openTransaction("Layers change")
+                    changed = True
+                obj.ViewObject.LineWidth = width
 
             # draw style
             style = self.model.item(row, 3).text()
-            if style:
-                if obj.ViewObject.DrawStyle != style:
-                    if not changed:
-                        FreeCAD.ActiveDocument.openTransaction("Layers change")
-                        changed = True
-                    obj.ViewObject.DrawStyle = style
+            if obj.ViewObject.DrawStyle != style:
+                if not changed:
+                    FreeCAD.ActiveDocument.openTransaction("Layers change")
+                    changed = True
+                obj.ViewObject.DrawStyle = style
 
             # line color
             color = self.model.item(row, 4).data(QtCore.Qt.UserRole)
-            if color:
-                if obj.ViewObject.LineColor[3:] != color:
-                    if not changed:
-                        FreeCAD.ActiveDocument.openTransaction("Layers change")
-                        changed = True
-                    obj.ViewObject.LineColor = color
```

Comment:
Don't we need this check for None?

Reply:
You have made the same comment here: #18989.

I am not convinced that this check is necessary. It may be standard practice though.

Reply:
Yes I did, I wanted you to see this so you didn't remove it by mistake, merge it unseen and introduce a regression because of a none check removal.

This could be totally fine, I don't know, but it deserved a comment 🙂